pallets/elections/src/impls.rs

```diff
@@ -293,6 +304,58 @@ where
 
         T::ValidatorRewardsHandler::add_rewards(rewards);
     }
+
+    fn calculate_underperforming_non_reserved_validators() {
+        let thresholds = CurrentEraCommitteeKickOutConfig::<T>::get();
+        let current_committee = T::SessionInfoProvider::current_committee();
+        let next_era_non_reserved = NextEraNonReservedValidators::<T>::get()
+            .into_iter()
+            .collect::<BTreeSet<_>>();
+        let non_reserved_nodes_in_next_era = current_committee.intersection(&next_era_non_reserved);
+        let expected_blocks_per_validator = Self::blocks_to_produce_per_session();
+        for validator in non_reserved_nodes_in_next_era {
+            let underperformance = match SessionValidatorBlockCount::<T>::try_get(&validator) {
+                Ok(block_count) => {
+                    let block_performance_ratio = match block_count <= expected_blocks_per_validator
+                    {
+                        true => Perbill::from_rational(block_count, expected_blocks_per_validator),
+                        false => Perbill::one(),
+                    };
+                    block_performance_ratio <= thresholds.minimal_expected_performance
```

Comment:
Isn't this equivalent to simply:
```rust
Perbill::from_rational(block_count, expected_block_per_validator) <= thresholds.minimal_expected_performance
```
If I'm reading [the code](https://github.com/paritytech/substrate/blob/e2fc0430ab3bcd644cded0353d4514611256ad3c/core/sr-primitives/src/lib.rs#L201) correctly, `n / m` (where n > m) will simplify to `Perbill::one()`.

Unless, again, this is on purpose to improve performance.

Reply:
Good point, let me change this

Reply:
It's done